### PR TITLE
Arcbox

### PR DESCRIPTION
**Pull request by dkirby-ms**

### CODE REVIEW

File: docs/azure_jumpstart_arcbox/_index.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
+# Jumpstart ArcBox - An Azure Arc Jumpstart project
+
+## Jumpstart ArcBox - Overview
+
+![ArcBox architecture diagram](./arch.png)
+
+Jumpstart ArcBox is a project that provides an easy to deploy sandbox for all things Azure Arc. ArcBox is designed to be completely self-contained within a single Azure subscription and resource group, which will make it easy for a user to get hands-on with all available Azure Arc technology with nothing more than an available Azure subscription.
+
+Azure Arc capabilities available in ArcBox include:
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:9]:
```suggestion
## Azure Arc capabilities available in ArcBox
```

File: docs/azure_jumpstart_arcbox/_index.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+### Azure Arc enabled Servers
+
+![ArcBox servers diagram](./servers.png)
+
+ArcBox includes three Azure Arc enabled server resources. These servers are hosted using nested virtualization in Azure. As part of the deployment, a Hyper-V host (ArcBox-Client) is deployed with three guest virtual machines. These machines, ArcBox-Win, ArcBox-Ubuntu, and ArcBox-SQL are connected as Azure Arc enabled servers via the ArcBox automation.
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:15]:
```suggestion
ArcBox includes three Azure Arc enabled server resources. These servers are hosted using nested virtualization in Azure. As part of the deployment, a Hyper-V host (ArcBox-Client) is deployed with three guest virtual machines. These machines, _ArcBoxWin_, _ArcBoxUbuntu_, and _ArcBoxSQL_ are connected as Azure Arc enabled servers via the ArcBox automation.
```

Suggest to keep current names until we will update

File: docs/azure_jumpstart_arcbox/_index.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+### Azure Arc enabled Kubernetes
+
+![ArcBox Kubernetes diagram](./k8s.png)
+
+ArcBox deploys one Rancher (K3s) cluster running on an Azure virtual machine. This cluster is then connected to Azure as an Azure Arc enabled Kubernetes resource (ArcBox-K3s).
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:21]:
```suggestion
ArcBox deploys one, single-node Rancher K3s cluster running on an Azure virtual machine. This cluster is then connected to Azure as an Azure Arc enabled Kubernetes resource (_ArcBox-K3s_).
```

File: docs/azure_jumpstart_arcbox/_index.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+    ```shell
+    az login
+    az ad sp create-for-rbac -n "<Unique SP Name>" --role contributor
+    ```
+
+    For example:
+
+    ```shell
+    az ad sp create-for-rbac -n "http://AzureArcGCP" --role contributor
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:57]:
```suggestion
    az ad sp create-for-rbac -n "http://AzureArcBox" --role contributor
```

File: docs/azure_jumpstart_arcbox/_index.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
+# Jumpstart ArcBox - An Azure Arc Jumpstart project
+
+## Jumpstart ArcBox - Overview
+
+![ArcBox architecture diagram](./arch.png)
+
+Jumpstart ArcBox is a project that provides an easy to deploy sandbox for all things Azure Arc. ArcBox is designed to be completely self-contained within a single Azure subscription and resource group, which will make it easy for a user to get hands-on with all available Azure Arc technology with nothing more than an available Azure subscription.
+
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:8]: Make consider adding short use-case section?

File: docs/azure_jumpstart_arcbox/_index.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+### Azure Arc enabled Kubernetes
+
+![ArcBox Kubernetes diagram](./k8s.png)
+
+ArcBox deploys one single-node Rancher K3s cluster running on an Azure virtual machine. This cluster is then connected to Azure as an Azure Arc enabled Kubernetes resource (_ArcBox-K3s_).
+
+### Azure Arc enabled Data Services
+
+ArcBox deploys an AKS cluster, which is then used to deploy an Azure Arc enabled data services data controller.
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:25]:
```suggestion
ArcBox deploys an AKS cluster, which is then being used to deploy an Azure Arc enabled data services data controller.
```

File: docs/azure_jumpstart_arcbox/_index.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+
+Jumpstart ArcBox is a project that provides an easy to deploy sandbox for all things Azure Arc. ArcBox is designed to be completely self-contained within a single Azure subscription and resource group, which will make it easy for a user to get hands-on with all available Azure Arc technology with nothing more than an available Azure subscription.
+
+## Azure Arc capabilities available in ArcBox
+
+### Azure Arc enabled Servers
+
+![ArcBox servers diagram](./servers.png)
+
+ArcBox includes three Azure Arc enabled server resources. These servers are hosted using nested virtualization in Azure. As part of the deployment, a Hyper-V host (ArcBox-Client) is deployed with three guest virtual machines. These machines, _ArcBoxWin_, _ArcBoxUbuntu_, and _ArcBoxSQL_ are connected as Azure Arc enabled servers via the ArcBox automation.
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:15]:
```suggestion
ArcBox includes three Azure Arc enabled server resources and are hosted using nested virtualization in Azure. As part of the deployment, a Hyper-V host (ArcBox-Client) is deployed with three guest virtual machines. These machines, _ArcBoxWin_, _ArcBoxUbuntu_, and _ArcBoxSQL_ are connected as Azure Arc enabled servers via the ArcBox automation.
```

File: docs/azure_jumpstart_arcbox/_index.md [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+
+ArcBox deploys an AKS cluster, which is then used to deploy an Azure Arc enabled data services data controller.
+
+![ArcBox data services diagram](./dataservices.png)
+
+## Automation flow
+
+![Deployment flow diagram](./deploymentflow.png)
+
+ArcBox uses an advanced automation flow to deploy and configure all necessary resources with minimal user interaction. The above diagram provides a high level overview of the deployment flow. A high level summary of the deployment is:
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:33]:
```suggestion
ArcBox uses an advanced automation flow to deploy and configure all necessary resources with minimal user interaction. The above diagram provides a high-level overview of the deployment flow. A high-level summary of the deployment is:
```

File: docs/azure_jumpstart_arcbox/_index.md [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+
+* User deploys the primary ARM template (azuredeploy.json). This template contains several nested templates that will run simultaneously.
+  * ClientVM ARM template - deploys the Client Windows VM
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:36]:
```suggestion
  * ClientVM ARM template - deploys the Client Windows VM. This is the Hyper-V host VM where all user interactions with the environment are made from. 
```

File: docs/azure_jumpstart_arcbox/_index.md [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+  * AKS ARM template - deploys AKS cluster which will be used to run Azure Arc enabled data services
+  * Rancher K3s template - deploys a Linux VM which will have Rancher (K3s) installed on it and connected as an Azure Arc enabled Kubernetes cluster
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:38]:
```suggestion
  * Rancher K3s template - deploys a Linux, Ubuntu VM which will have Rancher (K3s) installed on it and connected as an Azure Arc enabled Kubernetes cluster
```

File: docs/azure_jumpstart_arcbox/_index.md [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+  * Storage account template - used for staging files in automation scripts
+  * Management artifacts template - deploys Log Analytics workspace and solutions and Azure Policy artifacts
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:40]:
```suggestion
  * Management artifacts template - deploys Azure Log Analytics workspace and solutions and Azure Policy artifacts
```

File: docs/azure_jumpstart_arcbox/_index.md [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+* User remotes into Client Windows VM, which automatically kicks off multiple scripts that:
+  * Deploy and configure three (3) nested virtual machines in Hyper-V
+    * Windows VM - onboarded as Azure Arc enabled Server
+    * Ubuntu VM - onboarded as Azure Arc enabled Server
+    * Windows VM running SQL Server - onboarded as Azure Arc enabled SQL Server
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:45]:
```suggestion
    * Windows VM running SQL Server - onboarded as Azure Arc enabled SQL Server (as well as Azure Arc enabled Server)
```

File: docs/azure_jumpstart_arcbox/_index.md [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+  * Deploy and configure Azure Arc enabled data services on the AKS cluster including a data controller, a SQL MI instance, and a Postgres instance. After deployment, Azure Data Studio opens automatically with connection entries for each database instance. Data services deployed by the script are:
+    * Data controller
+    * SQL MI instance
+    * Postgres instance
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:49]: This part feels repetitive

File: docs/azure_jumpstart_arcbox/_index.md [hunk 14]
```diff
@@ -0,0 +1,159 @@
...
+  * AKS ARM template - deploys AKS cluster which will be used to run Azure Arc enabled data services
+  * Rancher K3s template - deploys a Linux VM which will have Rancher (K3s) installed on it and connected as an Azure Arc enabled Kubernetes cluster
+  * Storage account template - used for staging files in automation scripts
+  * Management artifacts template - deploys Log Analytics workspace and solutions and Azure Policy artifacts
+* User remotes into Client Windows VM, which automatically kicks off multiple scripts that:
+  * Deploy and configure three (3) nested virtual machines in Hyper-V
+    * Windows VM - onboarded as Azure Arc enabled Server
+    * Ubuntu VM - onboarded as Azure Arc enabled Server
+    * Windows VM running SQL Server - onboarded as Azure Arc enabled SQL Server
+  * Deploy and configure Azure Arc enabled data services on the AKS cluster including a data controller, a SQL MI instance, and a Postgres instance. After deployment, Azure Data Studio opens automatically with connection entries for each database instance. Data services deployed by the script are:
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:46]:
```suggestion
  * Deploy and configure Azure Arc enabled data services on the AKS cluster including a data controller, a SQL MI instance, and a PostgreSQL Hyperscale cluster. After deployment, Azure Data Studio opens automatically with connection entries for each database instance. Data services deployed by the script are:
```

File: docs/azure_jumpstart_arcbox/_index.md [hunk 15]
```diff
@@ -0,0 +1,159 @@
...
+    * Data controller
+    * SQL MI instance
+    * Postgres instance
+
+## Deployment Option 1: Azure Portal
+
+* Click the [![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fmicrosoft%2Fazure_arc%2Farcbox%2Fazure_jumpstart_arcbox%2Fazuredeploy.json) button and enter values for the the ARM template parameters.
+
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:54]: Add a screenshot of the Azure portal with the parameters view. I also recommend adding the "Deployment in progress" screen and "deployment completed" screen.

File: docs/azure_jumpstart_arcbox/_index.md [hunk 16]
```diff
@@ -0,0 +1,159 @@
...
+    "password": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+    "tenant": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+    }
+    ```
+
+    > **Note: It is optional but highly recommended to scope the SP to a specific [Azure subscription and resource group](https://docs.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest)**
+
+* Login to AZ CLI using the ```az login``` command.
+
+* Now you will need to edit the [azuredeploy.parameters.json](../../azure_jumpstart_arcbox/azuredeploy.parameters.json) ARM template parameters file and supply some values for your environment.
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:102]:
```suggestion
* Edit the [azuredeploy.parameters.json](../../azure_jumpstart_arcbox/azuredeploy.parameters.json) ARM template parameters file and supply some values for your environment.
```

File: docs/azure_jumpstart_arcbox/_index.md [hunk 17]
```diff
@@ -0,0 +1,159 @@
...
+  * *sshRSAPublicKey* - Your SSH public key
+  * *spnClientId* - Your Azure service principal id
+  * *spnClientSecret* - Your Azure service principal secret
+  * *spnTenantId* - Your Azure tenant id
+  * *windowsAdminUsername* - Client Windows VM Administrator name
+  * *windowsAdminPassword* - Client Windows VM administrator password
+  * *myIpAddress* - Your local IP address. This is used to allow remote RDP and SSH connections to the Client Windows VM and K3s Rancher VM.
+  * *logAnalyticsWorkspaceName* - Unique name for the ArcBox log analytics workspace
+
+    ![Screenshot showing example parameters](./parameters.png)
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:114]: make sure to nuke the SPN in the screenshot

File: docs/azure_jumpstart_arcbox/_index.md [hunk 18]
```diff
@@ -0,0 +1,159 @@
...
+  az group create --name <Name of the Azure resource group> --location <Azure Region>
+  az deployment group create \
+  --resource-group <Name of the Azure resource group> \
+  --template-file azuredeploy.json \
+  --parameters azuredeploy.parameters.json 
+  ```
+
+  ![Screenshot showing az group create](./azgroupcreate.png)
+
+  ![Screenshot showing az deployment group create](./azdeploy.png)
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:128]: mask subscription

File: docs/azure_jumpstart_arcbox/_index.md [hunk 19]
```diff
@@ -0,0 +1,159 @@
...
+
+* After deployment, you should see the ArcBox resources inside your resource group.
+
+  ![Screenshot showing az deployment group create](./deployedresources.png)
+
+* Open a remote desktop connection into ArcBox-Client. Upon logging in, multiple automated scripts will open and start running. These scripts will take 10-20 minutes to finish. Once completed, Azure Data Studio will start. At this point, the deployment is complete.
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:134]:
```suggestion
* Open a remote desktop connection into _ArcBox-Client_. Upon logging in, multiple automated scripts will open and start running. These scripts will take 10-20 minutes to finish and once completed, Azure Data Studio will start. At this point, the deployment is complete.
```

File: docs/azure_jumpstart_arcbox/_index.md [hunk 20]
```diff
@@ -0,0 +1,159 @@
...
+
+  ![Screenshot showing ArcBox-Client](./clientscript.png)
+
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:137]: I think there is a "Usage" section missing to show how to use the environment:

- Details on how to login the nested VM
- SQL Management studio in the SQL VM
- SQL Assessment in the portal
- No mention on kubectx and how to interact with the AKS/K3s clusters
- Azure Monitor for containers on the K3s Arc cluster (And AKS for the matter)
- azdata login to the controller
- etc

File: docs/azure_jumpstart_arcbox/_index.md [hunk 21]
```diff
@@ -0,0 +1,159 @@
...
+## Example ArcBox usage scenarios
+
+ArcBox is a sandbox that can be used for a large variety of use cases, such as a sandbox environment for testing and training or kickstarter for proof of concept projects. Some suggested next steps for you to try in your ArcBox are:
+
+* Deploy GitOps configurations on the K8s clusters
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:142]:
```suggestion
* Deploy GitOps configurations with Azure Arc enabled Kubernetes
```

File: docs/azure_jumpstart_arcbox/_index.md (hunk 20 above)
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:137]: Add a screenshot and highlight the Arc resources

File: docs/azure_jumpstart_arcbox/_index.md [hunk 22]
```diff
@@ -0,0 +1,159 @@
...
+* Build policy initiatives that apply to your Azure Arc enabled resources
+* Incorporate your own tooling and automation into the existing automation framework
+* Build a certificate/secret/key management strategy with your Azure Arc resources
+
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:146]: - Azure Policy?
- Key Vault?
- something else?

File: docs/azure_jumpstart_arcbox/_index.md [hunk 23]
```diff
@@ -0,0 +1,159 @@
...
+```shell
+az group delete -n <name of your resource group>
+```
+
+![Screenshot showing az group delete](./azdelete.png)
+
+## Known issues
+
+* Sample databases are not deployed into the MSSQL or Postgres instances at this time.
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:159]:
```suggestion
* Sample databases are not deployed into the MSSQL or PostgreSQL instances at this time.
```

File: docs/azure_jumpstart_arcbox/_index.md [hunk 24]
```diff
@@ -0,0 +1,159 @@
...
+## Clean up the deployment
+
+To clean up your deployment, simply delete the resource group.
+
+```shell
+az group delete -n <name of your resource group>
+```
+
+![Screenshot showing az group delete](./azdelete.png)
+
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:156]: add screenshot for Azure portal resource group delete

File: docs/azure_jumpstart_arcbox/release_notes/_index.md [hunk 25]
```diff
@@ -0,0 +1,12 @@
+
+# Jumpstart ArcBox release notes
+
+Release notes will be released on the first week of each month and will cover the previous month.
+
+## April 2021
+
+### Jumpstart ArcBox v0.1 release
+
+* This is the initial release for Jumpstart ArcBox, a project that provides an easy to deploy sandbox for all things Azure Arc. ArcBox is designed to be completely self-contained within a single Azure subscription and resource group, which will make it easy for a user to get hands-on with all available Azure Arc technology with nothing more than an available Azure subscription.
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/release_notes/_index.md:10]:
```suggestion
* This is the initial release for Jumpstart ArcBox, a project that provides an easy to deploy sandbox for all things Azure Arc. ArcBox is designed to be completely self-contained within a single Azure subscription and resource group, which will make it easy for a user to get hands-on with all available Azure Arc technologies with nothing more than an available Azure subscription.
```

File: docs/azure_jumpstart_arcbox/_index.md [hunk 26]
```diff
@@ -0,0 +1,159 @@
...
+  ![Screenshot showing az deployment group create](./deployedresources.png)
+
+* Open a remote desktop connection into ArcBox-Client. Upon logging in, multiple automated scripts will open and start running. These scripts will take 10-20 minutes to finish. Once completed, Azure Data Studio will start. At this point, the deployment is complete.
+
+  ![Screenshot showing ArcBox-Client](./clientscript.png)
+
+## Example ArcBox usage scenarios
+
+ArcBox is a sandbox that can be used for a large variety of use cases, such as a sandbox environment for testing and training or kickstarter for proof of concept projects. Some suggested next steps for you to try in your ArcBox are:
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:140]: Consider referencing (hyperlinks) to specific scenarios section in the Jumpstart

**Review by likamrat — CHANGES_REQUESTED** [review]
- Lingo updates
- Use-Cases section additions and positioning
- "How to use" section"
- Screenshots addtion
- Update screenshots

File: docs/azure_jumpstart_arcbox/_index.md (hunk 17 above)
Comment (dkirby-ms) [reply docs/azure_jumpstart_arcbox/_index.md:114]: this is not a real key

File: azure_jumpstart_arcbox/kubernetes/ubuntuRancher.json [hunk 27]
```diff
@@ -0,0 +1,361 @@
...
+            "type": "string",
+            "defaultValue": "ArcBox-K3s-NSG",
+            "metadata": {
+                "description": "Name of the Network Security Group"
+            }
+        },
+        "resourceTags": {
+            "type": "object",
+            "defaultValue": {
+                "Project": "jumpstart_azure_arc_k8s"
```
Comment (likamrat) [inline azure_jumpstart_arcbox/kubernetes/ubuntuRancher.json:79]:
```suggestion
                "Project": "jumpstart_arcbox"
```

File: docs/azure_jumpstart_arcbox/_index.md [hunk 28]
```diff
@@ -0,0 +1,188 @@
...
+  * Management artifacts template - deploys Azure Log Analytics workspace and solutions and Azure Policy artifacts
+* User remotes into Client Windows VM, which automatically kicks off multiple scripts that:
+  * Deploy and configure three (3) nested virtual machines in Hyper-V
+    * Windows VM - onboarded as Azure Arc enabled Server
+    * Ubuntu VM - onboarded as Azure Arc enabled Server
+    * Windows VM running SQL Server - onboarded as Azure Arc enabled SQL Server (as well as Azure Arc enabled Server)
+  * Deploy and configure Azure Arc enabled data services on the AKS cluster including a data controller, a SQL MI instance, and a PostgreSQL Hyperscale cluster. After deployment, Azure Data Studio opens automatically with connection entries for each database instance. Data services deployed by the script are:
+    * Data controller
+    * SQL MI instance
+    * Postgres instance
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:62]:
```suggestion
    * PostgreSQL Hyperscale cluster
```

File: docs/azure_jumpstart_arcbox/_index.md [hunk 29]
```diff
@@ -0,0 +1,188 @@
...
+
+  ![Screenshot showing az deployment group create](./deployedresources.png)
+
+* Open a remote desktop connection into _ArcBox-Client_. Upon logging in, multiple automated scripts will open and start running. These scripts will take 10-20 minutes to finish and once completed, Azure Data Studio will start. At this point, the deployment is complete.
+
+  ![Screenshot showing ArcBox-Client](./clientscript.png)
+
+  ![Screenshot showing ArcBox resources in Azure Portal](./rgarc.png)
+
+## Using ArcBox
```
Comment (likamrat) [inline docs/azure_jumpstart_arcbox/_index.md:159]: We need more explanations/details here on the tools installed on the client VM:

1. How does a user log in to the nested VMs? we need to provide the username/password in the guide.
2. How does a user switch between the AKS/k3s clusters using kubectx.
3. Show how to do _azdata login_ and the fact the CLI is there.

Without knowing these things they will have to dig in the scripts to figure it out.

**Review by likamrat — CHANGES_REQUESTED** [review]
Just a bit more flavor on how to use the tools/resources deployed with ArcBox.

**Comment by dkirby-ms** [comment]
Feedback incorporated. Thank you, please review.

**Review by likamrat — APPROVED** [review]
LGTM

**Pull request merged**